### PR TITLE
Added an enhancement - issue 366 to add a download all if no argument…

### PR DESCRIPTION
**Pull request by rabroldan**
This updated version automatically gets all blob fields from self.datasource.fields and adds them to the list of fields that will be downloaded if the fields argument is empty. As a result, the function will by default download all of the blob's fields if we don't specify any specific fields as parameters.

Adding an ```if condition``` to add the code that set the fields 

Kindly please rake a look thank you

### CODE REVIEW

File: dagshub/data_engine/model/query_result.py [hunk 1]
```diff
@@ -208,21 +208,26 @@ def __getitem__(self, item: Union[str, int, slice]):
...
             cache_on_disk: Whether to cache the blobs on disk or not (valid only if load_into_memory is set to True)
                 Cache location is `~/dagshub/datasets/<user>/<repo>/<datasource_id>/.metadata_blobs/`
         """
         send_analytics_event("Client_DataEngine_downloadBlobs", repo=self.datasource.source.repoApi)
         if not load_into_memory:
             assert cache_on_disk
+
+        # If no fields are specified, include all blob fields from self.datasource.fields
+        if not fields:
+            fields = [field.name for field in self.datasource.fields if field.is_blob()]
```
Comment (kbolashev) [inline dagshub/data_engine/model/query_result.py:229]: There is no `is_blob()` function on the `MetadataFieldSchema` enum.
You can either add this function, or compare to MetadataFieldType.BLOB in the comprehension

```suggestion
            fields = [field.name for field in self.datasource.fields if field.valueType == MetadataFieldType.BLOB]
```

Don't forget to import MetadataFieldType, I don't think it's imported in QueryResult.

**Review by kbolashev — CHANGES_REQUESTED** [review]
Looking good! Need to change a thing though.

Also apologies for not writing this anywhere like a CONTRIBUTING.md (we should probably set up one), but we use [black](https://github.com/psf/black) to format the code.

**Comment by rabroldan** [comment]
hi @kbolashev Kindly please review. I added and chaged the valeu as requested and added an import ```from dagshub.data_engine.dtypes import MetadataFieldType``` also reformatted the code using python black with the added codes into it.

**Comment by kbolashev** [comment]
Thank you so much @rabroldan !
Ironing out some stuff still, can I ask you to rerun black with `-l 120`?

I'll make sure to add it to `pyproject.toml`, and also clear it up in the whole repo so it's consistent.

**Comment by kbolashev** [comment]
Ok I reran it and resolved the merge conflict, should be able to merge it now.
Thank you for the contribution!

**Comment by rabroldan** [comment]
hey @kbolashev i jsut saw the request. but it seemed you already did it, thank you for allowing me to contribute on this project. i'll keep an eye on some of the issue and see if i can get a few more done.

**Pull request merged**